### PR TITLE
Cherry pick main 4.0.0 into stable-3

### PR DESCRIPTION
**Pull request by mikemorency**
##### SUMMARY
This PR is cherry picking missing commits from main into stable-3 prior to the next 3.X release

##### ISSUE TYPE
- Backport

##### ADDITIONAL INFORMATION
Let me know if this is not how things are normally backported. As far as I can tell, these are commits that the backport-3 label was not applied to (or missed) but should be included in the next 3.x minor release

### CODE REVIEW

File: .github/workflows/sanity-tests.yaml [hunk 1]
```diff
@@ -75,9 +76,9 @@ jobs:
           # Add new versions announced in
           # https://github.com/ansible-collections/news-for-maintainers in a timely manner,
           # consider dropping testing against EOL versions and versions you don't support.
-          - stable-2.14
```
Comment (mariolenz) [inline .github/workflows/sanity-tests.yaml:78]: breaking

File: changelogs/fragments/492-remove-support-ansible-2-14.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+breaking_changes:
+  - Removing any support for ansible-core <=2.14
```
Comment (mariolenz) [inline changelogs/fragments/492-remove-support-ansible-2-14.yml:2]: breaking

File: README.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ The collection includes the VMware modules and plugins supported by Ansible VMwa
 <!--start requires_ansible-->
 ## Ansible version compatibility
 
-This collection has been tested against following Ansible versions: **>=2.14.0**.
+This collection has been tested against following Ansible versions: **>=2.15.0**.
```
Comment (mariolenz) [inline README.md:10]: breaking

File: meta/runtime.yml [hunk 4]
```diff
@@ -1,135 +1,135 @@
-requires_ansible: '>=2.14.0'
+requires_ansible: ">=2.15.0"
```
Comment (mariolenz) [inline meta/runtime.yml:1]: breaking

File: tests/sanity/ignore-2.14.txt [hunk 5]
```diff
@@ -1,2 +0,0 @@
-plugins/modules/vcenter_vm_guest_customization.py pep8!skip
-plugins/modules/appliance_infraprofile_configs.py pep8!skip
```
Comment (mariolenz) [inline tests/sanity/ignore-2.14.txt:2]: breaking

**Review by mariolenz — CHANGES_REQUESTED** [review]
#495 has been a breaking change that requires a new major version. We can't backport this.

**Comment by mariolenz** [comment]
Additionally, should we also backport #516 while we're at it?

**Comment by mikemorency** [comment]
516 was added after the 4.0.0 release, does that matter? It's just a readme change so I'm guessing we may as well include it?

File: tests/sanity/ignore-2.14.txt (hunk 5 above)
Comment (mikemorency) [reply tests/sanity/ignore-2.14.txt:2]: Ah, right. I'll back these out

**Comment by mariolenz** [comment]
> 516 was added after the 4.0.0 release, does that matter? It's just a readme change so I'm guessing we may as well include it?

I think it would be a good idea, otherwise I wouldn't have suggested is ;-)

But maybe @bardielle or @machacekondra think differently. Personally, I don't think this is a problem.

**Comment by mikemorency** [comment]
OK, included 516 and removed the breaking change!

**Review by mariolenz — APPROVED** [review]
LGTM

**Comment by machacekondra** [comment]
So why not include the commits using the label?

**Comment by mikemorency** [comment]
i tried the label on two PRs and nothing happened. Maybe I am using it wrong? I applied it to my PRs https://github.com/ansible-collections/vmware.vmware_rest/pull/503 and https://github.com/ansible-collections/vmware.vmware_rest/pull/505

**Comment by machacekondra** [comment]
> i tried the label on two PRs and nothing happened. Maybe I am using it wrong? I applied it to my PRs #503 and #505

Yeah, seems it's broken for some time: https://github.com/ansible-collections/vmware.vmware_rest/actions/workflows/patchback.yml

**Review by machacekondra — APPROVED** [review]

**Pull request merged**